huaweicloud/services/acceptance/eg/data_source_huaweicloud_eg_event_sources_test.go

```diff
@@ -0,0 +1,249 @@
+					dcByChannelId.CheckResourceExists(),
+					resource.TestCheckOutput("is_channel_id_filter_useful", "true"),
+					dcByName.CheckResourceExists(),
+					resource.TestCheckOutput("is_name_filter_useful", "true"),
+					resource.TestCheckResourceAttrPair(byName, "sources.0.id", "huaweicloud_eg_custom_event_source.test", "id"),
+					resource.TestCheckResourceAttrPair(byName, "sources.0.channel_id", "huaweicloud_eg_custom_event_channel.test", "id"),
+					resource.TestCheckResourceAttrPair(byName, "sources.0.channel_name", "huaweicloud_eg_custom_event_channel.test", "name"),
+					resource.TestCheckResourceAttr(byName, "sources.0.name", name),
+					resource.TestCheckResourceAttr(byName, "sources.0.description", "Updated by terraform script"),
+					resource.TestCheckResourceAttr(byName, "sources.0.provider_type", "CUSTOM"),
+					resource.TestCheckResourceAttr(byName, "sources.0.type", "ROCKETMQ"),
+					resource.TestCheckResourceAttrSet(byName, "sources.0.status"),
+					resource.TestCheckResourceAttrSet(byName, "sources.0.created_at"),
+					resource.TestCheckResourceAttrSet(byName, "sources.0.updated_at"),
+					resource.TestCheckResourceAttrSet(byName, "sources.0.detail"),
+					dcByNotFoundName.CheckResourceExists(),
+					resource.TestCheckOutput("is_not_found_name_filter_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataEventSources_basic_base(name string) string {
+	return fmt.Sprintf(`
+data "huaweicloud_dms_rocketmq_instances" "test" {
+  instance_id = "%[1]s"
+}
+
+resource "huaweicloud_dms_rocketmq_consumer_group" "test" {
+  instance_id     = "%[1]s"
+  name            = "%[2]s"
+  enabled         = true
+  broadcast       = true
+  brokers         = ["broker-0"]
+  retry_max_times = 3
+
+  // Instances of version 5.x have no brokers returned, but versions 4.x have.
+  lifecycle {
+    ignore_changes = [
+      brokers
+    ]
+  }
+}
+
+resource "huaweicloud_dms_rocketmq_topic" "test" {
+  instance_id = "%[1]s"
+  name        = "%[2]s"
+  queue_num   = 3
+  permission  = "all"
+
+  brokers {
+    name = "broker-0"
+  }
+
+  lifecycle {
+    ignore_changes = [
+      brokers
+    ]
+  }
+}
+
+data "huaweicloud_vpc_subnets" "test" {
+  vpc_id = try(data.huaweicloud_dms_rocketmq_instances.test.instances[0].vpc_id, "")
+}
+
+resource "huaweicloud_eg_endpoint" "test" {
+  name      = "%[2]s"
+  vpc_id    = try(data.huaweicloud_dms_rocketmq_instances.test.instances[0].vpc_id, "")
+  subnet_id = try(data.huaweicloud_vpc_subnets.test.subnets[0].id, "")
+}
+
+resource "huaweicloud_eg_custom_event_channel" "test" {
+  depends_on = [huaweicloud_eg_endpoint.test]
+
+  name = "%[2]s"
+}
+`, acceptance.HW_DMS_ROCKETMQ_INSTANCE_ID, name)
+}
+
+func testAccDataEventSources_basic_step1(name string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_eg_custom_event_source" "test" {
+  channel_id = huaweicloud_eg_custom_event_channel.test.id
```

Comment:
```suggestion
  channel_id  = huaweicloud_eg_custom_event_channel.test.id
```
Aligning fields of scripts.

Reply:
done.